### PR TITLE
Update generic java

### PR DESCRIPTION
**Pull request by gOOvER**
# Description

- add all aviable java images
- convert to PTDL_v2
- update install image

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### CODE REVIEW

File: generic/java/egg-java.json [hunk 1]
```diff
@@ -25,7 +30,7 @@
     "scripts": {
         "installation": {
             "script": "#!\/bin\/bash\r\n# Java Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\nPROJECT=java\r\n\r\napt update\r\napt install -y curl jq\r\n\r\ncd \/mnt\/server",
-            "container": "debian:buster-slim",
+            "container": "ghcr.io\/parkervcp\/yolks:debian",
```
Comment (QuintenQVD0) [inline generic/java/egg-java.json:33]: Please don't use this installer image as it is not updated for arm64.

File: generic/java/egg-java.json [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
     "name": "Generic Java",
     "author": "sneaky@sneakyhub.com",
     "description": "Creates a container that runs java.",
     "features": null,
-    "images": [
-        "ghcr.io\/parkervcp\/yolks:java_8",
-        "ghcr.io\/parkervcp\/yolks:java_11",
-        "ghcr.io\/parkervcp\/yolks:java_16"
-    ],
+    "docker_images": {
```
Comment (QuintenQVD0) [inline generic/java/egg-java.json:12]: only 8,11,16,17,19 are sttill build: https://github.com/parkervcp/yolks/blob/master/.github/workflows/java.yml

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: generic/java/egg-java.json (hunk 1 above)
Comment (gOOvER) [reply generic/java/egg-java.json:33]: > Please don't use this installer image as it is not updated for arm64.

Thats not the installer ;)

File: generic/java/egg-java.json (hunk 2 above)
Comment (gOOvER) [reply generic/java/egg-java.json:12]: > only 8,11,16,17,19 are sttill build: [parkervcp/yolks@`master`/.github/workflows/java.yml](https://github.com/parkervcp/yolks/blob/master/.github/workflows/java.yml?rgh-link-date=2022-10-24T17%3A31%3A42Z)

Then someone should remove the old versions from the readme

File: generic/java/egg-java.json (hunk 1 above)
Comment (QuintenQVD0) [reply generic/java/egg-java.json:30]: ok I am blind :)

**Comment by QuintenQVD0** [comment]
Does it work? I remember normal images not working in the install container as there non root and can not install packages

**Comment by gOOvER** [comment]
> Does it work? I remember normal images not working in the install container as there non root and can not install packages

The install Image is for STEAM based games. The Normal Debian image is the same sa like: 

debian:bullseye_slim

**Comment by gOOvER** [comment]
see

https://github.com/parkervcp/yolks/blob/master/.github/workflows/base.yml

**Comment by QuintenQVD0** [comment]
I remember if I used that image as install image you get permisson denyed.  I will test if I have time tommorow

**Comment by gOOvER** [comment]
Then Something is wrong with the image

**Comment by QuintenQVD0** [comment]
It seems to work fine, but please use the new feature to display just the java version and not the full docker image url

**Comment by gOOvER** [comment]
> It seems to work fine, but please use the new feature to display just the java version and not the full docker image url

never know, that this is needed. I act after this:

https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md

The feature is nice, but you never see, what image is really needed

**Comment by QuintenQVD0** [comment]
> > It seems to work fine, but please use the new feature to display just the java version and not the full docker image url
> 
> never know, that this is needed. I act after this:
> 
> https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md
> 
> The feature is nice, but you never see, what image is really used

Yea true it is indeed not required but a little more clean to the end user. For me it is not needed if you don't want to implement it then it is fine for me in this way.

**Comment by gOOvER** [comment]
> > > It seems to work fine, but please use the new feature to display just the java version and not the full docker image url
> > 
> > 
> > never know, that this is needed. I act after this:
> > [`master`/CONTRIBUTING.md](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md?rgh-link-date=2022-10-25T16%3A03%3A57Z)
> > The feature is nice, but you never see, what image is really used
> 
> Yea true it is indeed not required but a little more clean to the end user. For me it is not needed if you don't want to implement it then it is fine for me in this way.

in my eye`s this is Hoster part and Contributor should not do the Hoster job. I only use it in this way and when it's wrong, so i cant commit anything other, because i dont use this feature

**Comment by gOOvER** [comment]
> > > It seems to work fine, but please use the new feature to display just the java version and not the full docker image url
> > 
> > 
> > never know, that this is needed. I act after this:
> > [`master`/CONTRIBUTING.md](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md?rgh-link-date=2022-10-25T16%3A03%3A57Z)
> > The feature is nice, but you never see, what image is really used
> 
> Yea true it is indeed not required but a little more clean to the end user. For me it is not needed if you don't want to implement it then it is fine for me in this way.

in my eye`s this is Hoster part and Contributor should not do the Hoster job. I only use it in this way and when it's wrong, so i cant commit anything other, because i dont use this feature

**Comment by QuintenQVD0** [comment]
> part

fine for me.

**Review by QuintenQVD0 — APPROVED** [review]

**Pull request merged**